scripts/cleanup_openstack_instance_snapshot.py

```diff
@@ -0,0 +1,63 @@
+#!/usr/bin/env python
+# -*- coding: utf-8 -*-
+"""Cleanup unused instance snapshot from glance repository
+
+Usage: scripts/cleanup_openstack_instance_snapshot.py [optional list of provider keys]
+
+If no providers specified, it will cleanup all of them.
+
+"""
+import sys
+from datetime import datetime
+from datetime import timedelta
+
+import iso8601
+import tzlocal
+
+from cfme.utils.log import add_stdout_handler
+from cfme.utils.log import logger
+from cfme.utils.providers import get_mgmt
+from cfme.utils.providers import list_provider_keys
+
+LOCAL_TZ = tzlocal.get_localzone()
+GRACE_TIME = timedelta(hours=2)
+TIME_LIMIT = datetime.now(tz=LOCAL_TZ) - GRACE_TIME
+
+# log to stdout too
+add_stdout_handler(logger)
+
+
+def main(*providers):
+    """ Cleanup all snapshots name starting with test_snapshot and created by >= 2 hours before
+
+    :param providers: Lists provider keys
+    :return:
+    """
+
+    for provider_key in providers:
+        logger.info("Cleaning up {}".format(provider_key))
+        provider_obj = get_mgmt(provider_key)
+
+        try:
+            images = provider_obj.list_templates()
+        except Exception as e:
+            logger.exception("Connect to provider failed:{} {} {}".format(
```

Comment:
You don't need to do all this, `logger.exception` logs the exception context. You just need to provide a plain language description of what the failure is.